File: Python/Leetcode Medium/128LongestConsecutiveSequence.py
#https://leetcode.com/problems/longest-consecutive-sequence/
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numsSet = set(nums)
        maxCounter = 0
        for num in nums:
            if num not in numsSet:
                continue
            tempCounter = 0
            curr = num
            while curr+1 in numsSet:
                tempCounter += 1
                curr += 1
                numsSet.remove(curr)
            curr = num
            while curr-1 in numsSet:
                tempCounter+=1 
                curr-=1 
                numsSet.remove(curr)
            maxCounter = max(maxCounter, tempCounter+1)
        return maxCounter

File: C++/Leetcode Medium/128LongestConsecutiveSequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> set;
        for(int i : nums) set.insert(i);
        int res = 0;
        for (int i : set){
            if (set.find(i-1) != set.end()) continue;
            int curr = i;
            int temp = 0;
            while (set.find(curr) != set.end()) {
                ++curr;
                ++temp;
            }
            res = max(res, temp);
        }
        return res;
    }
};
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> set;
        for (int i : nums) set.insert(i);
        int res = 0;
        for (int i : set) {
            if (set.find(i-1) == set.end()){
                int length = 0;
                while (set.find(i + length) != set.end()) ++length;
                res = max(res, length);
            }
        }
        return res;
    }
};